compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/window/DesktopPopup.desktop.kt

```diff
@@ -188,14 +189,22 @@ private fun PopupLayout(
 
                     layout(constraints.maxWidth, constraints.maxHeight) {
                         measurables.forEach {
-                            val placeable = it.measure(constraints)
-                            val position = popupPositionProvider.calculatePosition(
+                            var position = popupPositionProvider.calculatePosition(
+                                anchorBounds = parentBounds,
+                                windowSize = IntSize(width, height),
+                                layoutDirection = layoutDirection,
+                                popupContentSize = IntSize.Zero
+                            )
+                            // Vertically constraints correction
+                            val placeable = it.measure(constraints.copy(
+                                maxHeight = height - position.y
```

Comment:
We probably should take into account maxWidth too:
```
maxWidth = width - position.x
```

Reply:
Also, maybe we should take `max(height - position.y, position.y)`.

Imagine, a picture popup. If we open this popup at the bottom, the height of the picture would be too small, but there is a plenty of space above the popup.